### DISCUSSION TITLE
Dynamic Component Display and Custom URL Generation Based on Data in Column

### DISCUSSION THREAD

**Discussion by Winniepoom** [Budibase/budibase#11536] [Help]
### **_Updated 1and 2 of my issue is working now just the one in comment that I can figure out how to solve it_** 
Hello Budibase Team,

I've been using Budibase for a while now, and it has been incredibly helpful even though I'm not in a programming field. This tool has greatly assisted me in my work, and I want to express my gratitude to your team for that.

I do have a few questions regarding Budibase:

I'm interested in a function that controls the visibility of my card component based on the presence of data. Here's the scenario:

I have data in the 'issue_id' column (Formula type), and when I click on an employee in the table, it opens a right panel. Within this panel, there's a card component that displays the 'issue_id' – this works exactly as I intended.

![tempsnip](https://github.com/Budibase/budibase/assets/107398226/185e3e9f-4d45-4d7b-9d7d-b8d7618e2f0b)

However, when I click on an employee who doesn't have data in the 'issue_id' column, the card component appears blank with the text "card title." This isn't what I want. Ideally, I want the card component to remain hidden if the employee doesn't have data in the 'issue_id' column.

![image](https://github.com/Budibase/budibase/assets/107398226/1d514ba8-173c-497d-bf42-acf8f94b1dce)

### _******Now I managed to make it work by using Condition**_

![image](https://github.com/Budibase/budibase/assets/107398226/497ca8af-1413-40ea-aa96-7f35803d6455)

2. I would like to have a custom URL that is related to the 'issue_id' column. I want to be able to click on my card component and have it navigate to another website. For example, if I click on a card component for an employee who has a Redmine issue number of 558, I want it to navigate to http://localhost/issue/558. I want the number in the URL to be dynamic, depending on the 'issue_id' column. So, if the employee has an 'issue_id' number of 341, I want the number in the URL to change to that number.

### _*******Also this is working too**_  

Thank you for your assistance.

**Comment by Winniepoom** [comment]
Got Another Question:

I'm trying to split up my relationship data into a new column using this JS code:

```
var issueID0 = $("Redmine_Issue_API.0.issue_id");

if (issueID0 === undefined) {
    return "";
}

return issueID0;

```
It's all good when I use index 0, but when I switch to index number 1 like Redmine_Issue_API.1.issue_id, it gives me the wrong number. Check out the pic below – it should be showing '538' instead of '1963':

![image](https://github.com/Budibase/budibase/assets/107398226/f28b989d-d55a-4c4f-8433-9285044e06ac)

**Reply by ConorWebb96** [reply]
Hey @Winniepoom,

I'm having some issues visualising whats happening here. 

I'm assuming that the column with 1963 is a formula column. Are you wanting to get the second relationships id for it to be displayed in that column?

I've just tried getting the second attached id using the sample data from Budibase and didn't come across any issues. Below is the code as well as whats displayed. 
<img width="924" alt="Screenshot 2023-08-21 at 13 56 43" src="https://github.com/Budibase/budibase/assets/126772285/6f3e8f50-1241-4971-8b6b-51c48628fed5">
<img width="405" alt="Screenshot 2023-08-21 at 13 58 30" src="https://github.com/Budibase/budibase/assets/126772285/1d783881-f6d3-4cee-bdcc-2f1d629522b7">


Is there any chance I could get an app export so I can have a more in-depth look at what is being done? You can send this to **support@budibase.com**

Thanks ahead of time!

**Reply by Winniepoom** [reply]
@ConorWebb96 
Sorry that I'm not describing more precisely.
Yes, you understand correctly. The column '1963' is a formula column, and I want it to display the ID from the second relationship in that column.
Unfortunately, I cannot provide you with an exported app, as it contains real data, and sharing that would violate company policy.

So, I try to follow your step by using JS `return $ ("Redmine_Issue_API.1.issue_id")`  it show undefined and give me 1963
![image](https://github.com/Budibase/budibase/assets/107398226/4d4be084-1b41-4893-b726-4af3aaf3276d)

I noticed something different from what you explained. I'm not sure if this is related, but my relationship is of the text type:
![image](https://github.com/Budibase/budibase/assets/107398226/4c2fed56-9236-4bbd-b1fe-4efa217ce0a5)
